### ISSUE TITLE
[SplunkHECReceiver] Allow users to specify metadata in raw data 

### ISSUE THREAD

**Issue by splunkericl** [open-telemetry/opentelemetry-collector-contrib#19221]
**Is your feature request related to a problem? Please describe.**
To incorporate SplunkHECReceiver into EdgeProcessor(EP), the behavior of the raw endpoint should behave similar to how splunk enterprise endpoint is behaving. See [documentation](https://docs.splunk.com/Documentation/SplunkCloud/9.0.2209/Data/HTTPEventCollectortokenmanagement#:~:text=3...%20Hello%2C%20world!%27-,Send%20raw%20batched%20events%20to%20HEC,-The%20following%20example). 

For example, the users should be able to call raw endpoint like this to provide metadata like `channel`, `sourcetype`, `source` and `index`:
```
https://http-inputs-mysplunkserver.splunkcloud.com:8088/services/collector/raw?channel=00872DC6-AC83-4EDE-8AFE-8413C3825C4C&sourcetype=splunkd_access&index=main
```

**Describe the solution you'd like**
Extends the current endpoint of the rawendpoint to accept some parameters. Some important ones are:
- source
- sourcetype
- index

**Describe alternatives you've considered**
N/A

**Comment by timannguyen** [comment]
i will working on this issue.

**Comment by codeboten** [comment]
This has been addressed by #19632
